docker/py/entrypoint.sh

```diff
@@ -24,11 +24,25 @@ find /tmp/renku-env -not -path '*.git*' -type f -print0 | xargs --null -I{} sh -
 
 if [[ -v CI_PROJECT ]];
 then
+    # set the global git credentials
```

Comment:
This relies on what's happening in the init container. Maybe create the actual credentials here (again) and place them directly in the users home directory (ie outside of the repo)? This would make it a bit nicer to re-use the image in a non-k8s setting, but at the cost of having the same credentials in two places... In principle the init container could even remove the local credentials again after successful clone to avoid this duplication.

Reply:
Actually the best might be to check for local credentials here from the init container and remove them if they exist and then create the global ones.

Reply:
I wanted to limit the places where we inject the oauth token into the environment. It would be easy for users to leak their oauth token if they e.g. copy-pasted their environment variables somewhere. They're more likely to look at their environment variables than their `.git/config` file, but I agree with you that it's not ideal the way it's implemented.

Reply:
Ok, my starting point was that I'd expect local config in the repo and global config in the users home directory, but I see your point: it's true that the environment variable of the init-container are less exposed than those of the main container. So let's merge this.